src/core/gui/inputdevices/TouchInputHandler.cpp

```diff
@@ -22,64 +22,60 @@ TouchInputHandler::TouchInputHandler(InputContext* inputContext): AbstractInputH
 auto TouchInputHandler::handleImpl(InputEvent const& event) -> bool {
     bool zoomGesturesEnabled = inputContext->getSettings()->isZoomGesturesEnabled();
 
-    // Don't handle more then 2 inputs
-    if (this->primarySequence && this->primarySequence != event.sequence && this->secondarySequence &&
-        this->secondarySequence != event.sequence) {
-        return false;
-    }
-
     if (event.type == BUTTON_PRESS_EVENT) {
-        // Start scrolling when a sequence starts and we currently have none other
-        if (this->primarySequence == nullptr && this->secondarySequence == nullptr) {
-            this->primarySequence = event.sequence;
-
-            // Set sequence data
-            sequenceStart(event);
-        }
-        // Start zooming as soon as we have two sequences.
-        else if (this->primarySequence && this->primarySequence != event.sequence &&
-                 this->secondarySequence == nullptr) {
-            this->secondarySequence = event.sequence;
-
-            // Set sequence data
-            sequenceStart(event);
-
-            // Even if zoom gestures are disabled,
-            // this is still the start of a sequence. Just
-            // don't start zooming.
-            if (zoomGesturesEnabled) {
-                zoomStart();
+        if (invalidActive.empty() && event.sequence != primarySequence && event.sequence != secondarySequence) {
```

Comment:
With `invalidActive` never being cleared (besides in `unblock()` which does not really matter for this), the touchscreen will become unresponsive if we somehow miss a `BUTTON_RELEASE_EVENT`.
I've seen such events being skipped for pen devices, and I assume some touch drivers might have this defect as well.
Essentially, we cannot assume that the events come in a sequence BUTTON_PRESS - MOTION - ... - BUTTON_RELEASE.
(As a side note: this kind of undue trust in the drivers is present all over our input system and we need to strengthen it)

I'm not quite sure what a good solution might be in this specific case though